### ISSUE TITLE
Propper interface names in "show vlan"

### ISSUE THREAD

**Issue by rrendec** [lisa-project/lisa-user#1]
Currently raw netdevice name is listed. Special interface types (such as ethernet and portchannel) need to be listed in short canonical format (e.g. "Et1", "Po1" etc). This should be implemented in a function similar to canonical_if_name() in cfgbuild.c (perhaps both functions need to be moved to a library file).

Interface names also need to be sorted before they are printed in vlan list.

**Comment by acrish** [comment]
Do you refer to this part of the code?

```
if ((n = if_parse_ethernet(nsdev->name)) >= 0)
            status = asprintf(&ret, "Ethernet %d", n);
else
            status = asprintf(&ret, "netdev %s", nsdev->name);
```

Regarding sorting the interface names, whenever canonical_if_name() is called, the results should be kept in a data structure, sorted and eventually printed in vlan list, is that correct?

**Comment by rrendec** [comment]
Yes, basically that's the part of code that I have in mind. The action plan that I'm proposing is this:
- move canonical_if_name() from cli/cfgbuild/cfgbuild.c to lib/if_generic.c, since it may be useful as well for other modules than the config builder
- create a similar function, let's say short_if_name() that uses "Et%d" and "net %s" for that part of the code that you quoted

Now it's debatable if it's better to create a completely separate function or add a new parameter to canonical_if_name() that tells whether the result should be in long or short format.

The only counter-argument I can think of is that the same code structure will be replicated in a new function and if something needs to be changed, then it will need to be changed in two places. However, I think there are a little bit more advantages to using a separate function:
- code will be easier to read, since each branch of the code will not be clobbered with checking which format (long or short) needs to be returned
- for short names, there may be different cases cases and choices in the future, when we add more interface types; this means that the code flow will be different for long and short names, and therefore keeping the code unified will not make much sense

Going back to cmd_show_vlan(), the loop "for (j = 0; j < vlif_no; j++)" will have to call short_if_name() and put the result into a buffer instead of doing the comma_buffer_append() trick. Then the buffer needs to be sorted. Eventually, a new loop should iterate through the buffer and print out the results using comma_buffer_append()/print_buf/comma_buffer_reset().

It would be great if you could find a better approach than defining the print_buf macro in the middle of the function. It looks quite ugly as it is right now.

**Comment by rrendec** [comment]
Fixed in 4069be71718b3dc4c9fb0532a3c57f83854f0e2c.
